README.md

```diff
@@ -18,21 +17,99 @@ ember-es6-class-codemod <TRANSFORM NAME> path/of/files/ or/some**/*glob.js
 
 ## Transforms
 
-<!--TRANSFORMS_START-->
-<!--TRANSFORMS_END-->
+The transforms are tiered starting with simple ember objects to the complex ones including decorators and/or mixins etc. The codemods can be run targeting only specific types. For example it is possible to run the codemods targeting only `services` or `routes` in the app code.
+
+The codemods are grouped as following
+
+1. Simple objects (codemod: `simple-objects`)
+2. Services (codemod: `services`)
+3. Class helpers (codemod: `class-helpers`)
+4. Routes (codemod: `routes`)
+5. Components (codemod: `components`)
+6. Controllers (codemod: `controllers`)
+7. Routers (codemod: `routers`)
```

Comment:
I'm not sure we should be tiering based on type of object, but instead based on features required for a given class to be transformed. Any one of these classes could require decorators, mixins, or class fields, so it's not enough to delineate by class type. 

Additionally, it'll be hard to figure out what category subclasses belong to, and this strategy could result in half transformed files where one class is transpiled, but the other isn't.

I think instead we should add support for globbing, where users can define a _folder_ or _pattern_ that they want to apply the transforms to. This means that we always transform a whole file, no matter how many classes are part of it, but it also allows us to transform roughly by category (e.g. to transform all Routes, you would pass `--include "routes/**/*"`).

Reply:
The `jscodeshift` has support for glob. As per the issue here: https://github.com/facebook/jscodeshift/issues/215 it relies on the shell to expand the glob. So I don't think any thing is needed from the codemods to support that, since the files passed to `transform` function will already be matched against the path (or pattern) passed.

I like the idea of transforming by types, but as you mentioned it might be difficult or we might be in a state where part the file can be transformed and rest can not.

The current state of codemods is designed with command line options:
1. Transform simple objects (no options needed)
2. Transform object with decorators (Use `--decorators=true`)
3. Transform object with mixins (Use `--mixins=true`)
4. Transform object with decorators and mixins (Use `--mixins=true` and `--decorators=true`)

To support types - my initial approach would be to add additional parameter `--type=Controller|Route|Service|..` Using this parameter we can add different transform, for example `route-transforms` would have `--type=route` set by default. I haven't fully explored this approach yet, but I see there might be some cases where it is difficult to identify the type of the object.

Reply:
Added https://github.com/scalvert/ember-es6-class-codemod/pull/12, it matches the path to resolve the type.